scripts/build-docker-image.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+
+
+set -euxo pipefail
+
+ARCH=${ARCH:-$(uname -m)}
+
+# Set the correct build target and update the arch if required.
+if [[ "$ARCH" = "amd64" ]]; then
+  BUILD_TARGET="x86_64-unknown-linux-gnu"
+elif [[ "$ARCH" = "arm64" ]]; then
+  BUILD_TARGET="aarch64-unknown-linux-gnu"
+elif [[ "$ARCH" = "aarch64" ]]; then
+  BUILD_TARGET="aarch64-unknown-linux-gnu"
+  ARCH="arm64"
+fi
```

Comment:
Let's add an `else` branch that errors if no supported architecture is set, for instance:

```suggestion
if [[ "$ARCH" == "aarch64 ]]; then
  ARCH="arm64"
fi

if [[ "$ARCH" = "amd64" ]]; then
  BUILD_TARGET="x86_64-unknown-linux-gnu"
elif [[ "$ARCH" = "arm64" ]]; then
  BUILD_TARGET="aarch64-unknown-linux-gnu"
else
  echo "ERROR Unsupported architecture"
  exit 1
fi
```

Reply:
@jan-auer  changed to switch this one